includes/backend.inc

```diff
@@ -1152,15 +1150,15 @@ function _drush_backend_generate_command($site_record, $command, $args = array()
 
   $cmd[] = $command;
   foreach ($args as $arg) {
-    $cmd[] = drush_escapeshellarg($arg, $remote_os);
+    $cmd[] = drush_escapeshellarg($arg, $os);
```

Comment:
I think this change breaks backwards compatibility.

Reply:
I reverted a number of changes I had started with.  The final result should only be a change in `drush_is_windows()`.  I can try creating a new branch/PR to clean this up.